tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/PitSummaryView.java

```diff
@@ -16,34 +16,86 @@
 
 package org.pitest.pitclipse.ui.behaviours.pageobjects;
 
-import org.pitest.pitclipse.ui.behaviours.StepException;
-import org.pitest.pitclipse.ui.swtbot.PitNotifier;
-import org.pitest.pitclipse.ui.swtbot.PitResultsView;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import static org.pitest.pitclipse.ui.view.PitView.BACK_BUTTON_TEXT;
+import static org.pitest.pitclipse.ui.view.PitView.FORWARD_BUTTON_TEXT;
+import static org.pitest.pitclipse.ui.view.PitView.HOME_BUTTON_TEXT;
 
 public class PitSummaryView {
+    private static String SUMMARY_VIEW_TITLE = "PIT Summary";
+    private SWTWorkbenchBot bot;
+    private SWTBotView summaryView;
 
-    private PitResultsView lastResults = null;
-
-    public PitSummaryView() {
+    public PitSummaryView(SWTWorkbenchBot bot) {
+        this.bot = bot;
     }
 
-    public void waitForUpdate() {
-        try {
-            lastResults = PitNotifier.INSTANCE.getResults();
-        } catch (InterruptedException e) {
-            throw new StepException(e);
+    public void showViewIfNotOpen() {
+        if (summaryView == null) {
+            summaryView = bot.viewByTitle(SUMMARY_VIEW_TITLE);
```

Comment:
Do we really need a bot? I think we could get a reference to `PitView` instead.

Moreover, I think that the methods `goBack()`, `goForward()`, `goHome()` and `getUrl()` could be implemented directly in the `PitView` class.

Reply:
Sure, but why is this better than using the bot?

But I will implement it anyway now.

Reply:
And I also need the bot to change the url with JavaScript in order to be able to use relative urls.

I would prefer a different solution than JavaScript, but I didn't figure one out yet. < Probably a reason why the tests fail sometimes.

Reply:
Thank you very much for the change!

> Sure, but why is this better than using the bot?

That brings better cohesion because the methods that modify the _Mutations_ view are now in the class that is responsible of this view. Some advantages I can think of are that:

 - execution is likely (a bit) quicker as we directly trigger an action instead of asking a slow bot to find then click on a button,
 - `PitSummaryView` is no longer dependent on the _Mutations_ view's UI (the bot was assuming that the view was providing its actions as toolbar buttons, which may not be the case in the future),
 - if someone (can be us) wants to reuse `PitView` in the future then they can call, e.g., `back()` or `forward()` right away without having to implement another bot.

Reply:
Thanks for the explanation. Is this resolved?

Reply:
@echebbi sorry if I'm late, but if I understand correctly what you say I think it would be better to actually click on the buttons with the bot, after all, we are testing the UI, aren't we? Or I don't understand what you were suggesting.

Reply:
I'm open to change this back to use the bot,
Even both ways would be an option for me.

Reply:
@LorenzoBettini you're absolutely right, I completely overlooked the fact that this class aims at testing UI. So, yes, in this context we do want to click on the buttons... 

@JKutscha I'm sorry for the trouble but could you revert back to using the bot? It indeed makes sense to use it here.

Reply:
Alright, will do

Reply:
@echebbi Should I leave the methods like `back()` in place or delete them?

Reply:
You can delete them. We don't use them right now so they would basically be dead code to maintain.